### PR TITLE
feat: ensure all resourceVersions referenced by events are sent downstream

### PR DESCRIPTION
**Pull request by mmanciop**
Description: Patched the `k8sobjectsreceiver` to make it try to collect and send downstream all `resourceVersion`s of objects it sees referenced in events, and to ensure that the latest `resourceVersion` of objects referenced by `ownerReference`s are sent downstream ASAP on startup.

### CODE REVIEW

File: receiver/k8sobjectsreceiver/receiver.go [hunk 1]
```diff
@@ -190,3 +240,202 @@ func NewTicker(repeat time.Duration) *time.Ticker {
...
+	kr.setting.Logger.Debug("Retrieving object reference", zap.Any("objectReference", objectRef))
+
+	listOptions := metav1.ListOptions{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: objectRef.APIVersion,
+			Kind:       objectRef.Kind,
+		},
+	}
+
+	// We might not have a resource version, if this object reference is created from an owner reference
```
Comment (GuyMoses) [inline receiver/k8sobjectsreceiver/receiver.go:377]: shouldn't the owner reference has point to a resource version?

File: receiver/k8sobjectsreceiver/receiver.go [hunk 2]
```diff
@@ -190,3 +240,202 @@ func NewTicker(repeat time.Duration) *time.Ticker {
...
+	case deploymentv1:
+		{
+			var deployment appsv1.Deployment
+			if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructured.UnstructuredContent(), &deployment); err != nil {
+				return nil, fmt.Errorf("cannot parse appsv1.Deployment from unstructured: %w", err)
+			}
+
+			kr.addCacheEntry(deployment.APIVersion, deployment.Kind, string(deployment.ObjectMeta.UID), deployment.ObjectMeta.ResourceVersion)
+
+			return kr.ensureOwnerReferencesAreKnown(ctx, resource, &deployment.ObjectMeta.OwnerReferences)
```
Comment (GuyMoses) [inline receiver/k8sobjectsreceiver/receiver.go:289]: what if the deployment doesn't have an OwnerReference?

File: receiver/k8sobjectsreceiver/receiver_references_test.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		time.Sleep(time.Second * 1)
+
+		Eventually(func(g Gomega) {
+			logRecords := getLogRecords(consumer.Logs())
+
+			g.Expect(logRecords).Should(ContainElement(matchWatchEvent(watch.Added, toObjectReference(pod1))))
+			g.Expect(logRecords).Should(ContainElement(matchWatchEvent(watch.Added, toObjectReference(pod2))))
+		}, 1*time.Minute, 1*time.Second).Should(Succeed())
+
+		// TODO Improve tests using a deployment and check that all needed resourceVersions of pods and replicasets
```
Comment (GuyMoses) [inline receiver/k8sobjectsreceiver/receiver_references_test.go:112]: 😳

File: receiver/k8sobjectsreceiver/receiver.go (hunk 2 above)
Comment (mmanciop) [reply receiver/k8sobjectsreceiver/receiver.go:289]: I actually never saw a deployment that does have owner references, tbh. But, in that case, the `corev1.Deployment` object has an empty array as owner references. This is all thoroughly tested in the upcoming operator PR.

File: receiver/k8sobjectsreceiver/receiver_references_test.go (hunk 3 above)
Comment (mmanciop) [reply receiver/k8sobjectsreceiver/receiver_references_test.go:112]: I have not focused on the unit tests in the collector, I focused on the end-to-end tests of the `lumigo-kubernetes-operator` :D

File: receiver/k8sobjectsreceiver/receiver.go (hunk 1 above)
Comment (mmanciop) [reply receiver/k8sobjectsreceiver/receiver.go:377]: Nope: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#ownerreference-v1-meta

**Comment by mmanciop** [comment]
The failing integration test is for a receiver we do not care about.

**Pull request merged**